### PR TITLE
 refactor(connector): [Worldline][Worldpay][Zen] Mask PII data

### PR DESCRIPTION
**Pull request by AkshayaFoiger**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Mask pii information passed and received in the connector request and response for Worldline, Worldpay and zen.

## Test Case

_Note: Zen Googlepay and Worldpay can't be tested in sandbox_

1. Create a card payment with worldpay

2. Check if all the sensitive data in the `masked_response` is masked

```
curl --location '{{base_url}}/analytics/v1/connector_event_logs?type=Payment&payment_id={{payment_id}}' \
--header 'sec-ch-ua: "Not A(Brand";v="99", "Google Chrome";v="121", "Chromium";v="121"' \
--header 'sec-ch-ua-mobile: ?0' \
--header 'authorization: Bearer JWT_token' \
--header 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36' \
--header 'Content-Type: application/json' \
--header 'Referer: https://integ.hyperswitch.io/' \
--header 'api-key: {{api-key}}' \
--header 'sec-ch-ua-platform: "macOS"'
```

Response 
```
request\":\"{\\\"cardPaymentMethodSpecificInput\\\":{\\\"card\\\":{\\\"cardNumber\\\":\\\"456735**********\\\",\\\"cardholderName\\\":\\\"*** alloc::string::String ***\\\",\\\"cvv\\\":\\\"*** alloc::string::String ***\\\",\\\"expiryDate\\\":\\\"*** alloc::string::String ***\\\"},\\\"requiresApproval\\\":false,\\\"paymentProductId\\\":1},\\\"order\\\":{\\\"amountOfMoney\\\":{\\\"amount\\\":20000,\\\"currencyCode\\\":\\\"PLN\\\"},\\\"customer\\\":{\\\"billingAddress\\\":{\\\"city\\\":\\\"San Fransico\\\",\\\"countryCode\\\":\\\"US\\\",\\\"houseNumber\\\":null,\\\"state\\\":\\\"*** alloc::string::String ***\\\",\\\"stateCode\\\":null,\\\"street\\\":null,\\\"zip\\\":\\\"*** alloc::string::String ***\\\"},\\\"contactDetails\\\":{\\\"emailAddress\\\":\\\"*********@gmail.com\\\",\\\"mobilePhoneNumber\\\":null}},\\\"references\\\":{\\\"merchantReference\\\":\\\"pay_Cm4XoGAkTeXdAaAx3OUo_1\\\"}},\\\"shipping\\\":{\\\"city\\\":\\\"San Fransico\\\",\\\"countryCode\\\":\\\"US\\\",\\\"houseNumber\\\":null,\\\"name\\\":{\\\"firstName\\\":\\\"*** alloc::string::String ***\\\",\\\"surname\\\":\\\"*** alloc::string::String ***\\\",\\\"surnamePrefix\\\":null,\\\"title\\\":null},\\\"state\\\":\\\"*** alloc::string::String ***\\\",\\\"stateCode\\\":null,\\\"street\\\":null,\\\"zip\\\":\\\"*** alloc::string::String ***\\\"}}\",\"masked_response\":\"{\\\"payment\\\":{\\\"id\\\":\\\"000000113200000139460000100001\\\",\\\"status\\\":\\\"CAPTURE_REQUESTED\\\",\\\"capture_method\\\":\\\"automatic\\\"},
```

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by srujanchikke — DISMISSED** [review]

File: crates/router/src/connector/zen/transformers.rs [hunk 1]
```diff
@@ -201,7 +201,7 @@ pub struct SessionObject {
 #[derive(Debug, Serialize, Deserialize)]
 pub struct WalletSessionData {
     pub terminal_uuid: Option<String>,
-    pub pay_wall_secret: Option<String>,
+    pub pay_wall_secret: Option<Secret<String>>,
```
Comment (SamraatBansal) [inline crates/router/src/connector/zen/transformers.rs:204]: `terminal_uuid` also needs to be a secret

File: crates/router/src/connector/worldpay/requests.rs [hunk 2]
```diff
@@ -6,15 +6,15 @@ pub struct BillingAddress {
...
-    pub address2: Option<String>,
-    pub postal_code: String,
+    pub address2: Option<Secret<String>>,
+    pub postal_code: Secret<String>,
     #[serde(skip_serializing_if = "Option::is_none")]
     pub state: Option<String>,
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub address3: Option<String>,
+    pub address3: Option<Secret<String>>,
     pub country_code: String,
```
Comment (SamraatBansal) [inline crates/router/src/connector/worldpay/requests.rs:15]: `country_code`'s type can be changed from `String` to our `Currency` enum.

Accordingly update at all the places if necessary.

File: crates/router/src/connector/worldpay/response.rs [hunk 3]
```diff
@@ -218,11 +221,11 @@ impl PaymentInstrumentCardIssuer {
...
 pub struct PaymentInstrumentCardNumber {
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub bin: Option<String>,
+    pub bin: Option<Secret<String>>,
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub last4_digits: Option<String>,
+    pub last4_digits: Option<Secret<String>>,
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub dpan: Option<String>,
+    pub dpan: Option<Secret<String>>,
```
Comment (SamraatBansal) [inline crates/router/src/connector/worldpay/response.rs:228]: These are not required to be masked, please take @ArjunKarthik 's view on this once

**Review by SamraatBansal — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/worldpay/requests.rs (hunk 2 above)
Comment (AkshayaFoiger) [reply crates/router/src/connector/worldpay/requests.rs:15]: Field `country_code` under Submerchant, is expected to have a numeric value according to the [connector docs](https://developer.worldpay.com/docs/access-worldpay/api/references/payments/authorizations). Hence not updating the type of this field.

**Review by SamraatBansal — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Pull request merged**